kratos/tests/cpp_tests/sources/test_filesystem.cpp

```diff
@@ -191,5 +191,53 @@ KRATOS_TEST_CASE_IN_SUITE(FileSystemRenameDirectory, KratosCoreFastSuite)
     KRATOS_CHECK_IS_FALSE(Kratos::filesystem::exists(dir_name_new));
 }
 
+KRATOS_TEST_CASE_IN_SUITE(ListDirectory, KratosCoreFastSuite)
+{
+    const std::string dir_name("GetListOfDirectories");
+    const std::string dir_name_2("GetListOfDirectoriesSub");
+
+    const std::string raw_file_name_1("dummy_file_1.txt");
+    const std::string raw_file_name_2("dummy_file_2.txt");
+    const std::string sub_dir(Kratos::FilesystemExtensions::JoinPaths({dir_name, dir_name_2}));
+    const std::string file_name_1(Kratos::FilesystemExtensions::JoinPaths({dir_name, raw_file_name_1}));
+    const std::string file_name_2(Kratos::FilesystemExtensions::JoinPaths({dir_name, raw_file_name_2}));
+
+    KRATOS_CHECK_IS_FALSE(Kratos::filesystem::exists(dir_name));
+    KRATOS_CHECK(Kratos::filesystem::create_directory(dir_name));
+    KRATOS_CHECK_IS_FALSE(Kratos::filesystem::exists(sub_dir));
+    KRATOS_CHECK(Kratos::filesystem::create_directory(sub_dir));
+
+    std::ofstream output_file;
+    output_file.open(file_name_1);
+    KRATOS_CHECK(output_file.good());
+    output_file.close();
+
+    output_file.open(file_name_2);
+    KRATOS_CHECK(output_file.good());
+    output_file.close();
```

Comment:
why do you check this with opening the file?
shouldn't the `exists` test be enough?

Reply:
i will remove it... its a copy paste thing :/

Reply:
done :)